prov/udp/src/udpx_init.c

```diff
@@ -37,20 +37,26 @@
 
 #include <sys/types.h>
 
+static ofi_mutex_t init_lock;
 
 static int udpx_getinfo(uint32_t version, const char *node, const char *service,
 			uint64_t flags, const struct fi_info *hints,
 			struct fi_info **info)
 {
-	return ofi_ip_getinfo(&udpx_util_prov, version, node, service, flags,
-			      hints, info);
+	ofi_mutex_lock(&init_lock);
+	udpx_util_prov_init(version, node, service, flags);
+	ofi_mutex_unlock(&init_lock);
+	return util_getinfo(&udpx_util_prov, version, node, service, flags,
+			    hints, info);
```

Comment:
udpx_util_prov is initialized once with the "node", "service" and "flags" from the first fi_getinfo() call.  That likely won't work properly for later fi_getinfo calls with different parameters.

Reply:
Of course. Please check the updated code, which should address this.